### PR TITLE
Adding ability to list attestors

### PR DESCRIPTION
**Pull request by ChaosInTheCRD**
## What this PR does / why we need it

https://github.com/in-toto/witness/issues/372 highlights the lack of ability for users to list the attestors that are available for calling when running `witness run`.

This PR adds a `witness attestors` subcommand, which allows users to get a list of the attestors available back. In the future we should consider adding descriptions into the attestors in the https://github.com/in-toto/go-witness library and exposing those descriptions here.

Executing the command gives an output like:

```
➜ go run . attestors           
+-----------------------+---------------------------------------------------+-------------+
|         NAME          |                       TYPE                        |   RUNTYPE   |
+-----------------------+---------------------------------------------------+-------------+
| material (always run) | https://witness.dev/attestations/material/v0.1    | material    |
| gcp-iit               | https://witness.dev/attestations/gcp-iit/v0.1     | prematerial |
| github                | https://witness.dev/attestations/github/v0.1      | prematerial |
| gitlab                | https://witness.dev/attestations/gitlab/v0.1      | prematerial |
| git (default)         | https://witness.dev/attestations/git/v0.1         | prematerial |
| jwt                   | https://witness.dev/attestations/jwt/v0.1         | prematerial |
| oci                   | https://witness.dev/attestations/oci/v0.1         | postproduct |
| environment (default) | https://witness.dev/attestations/environment/v0.1 | prematerial |
| maven                 | https://witness.dev/attestations/maven/v0.1       | prematerial |
| aws                   | https://witness.dev/attestations/aws/v0.1         | prematerial |
| sarif                 | https://witness.dev/attestations/sarif/v0.1       | postproduct |
| product (always run)  | https://witness.dev/attestations/product/v0.1     | product     |
+-----------------------+---------------------------------------------------+-------------+
```

### CODE REVIEW

File: cmd/attestors.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	for _, entry := range entries {
+		name := entry.Factory().Name()
+		// NOTE: This is a workaround to avoid printing the command-run attestor. We should mark it in the registry as such somehow.
+		if name == "command-run" {
+			continue
+		}
+
+		for _, a := range alwaysRunAttestors {
+			if name == a.Name() {
+				name = name + " (always run)"
```
Comment (kriscoleman) [inline cmd/attestors.go:56]: this is very helpful for the distinction on product and material attestors.

File: cmd/attestors.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+			}
+		}
+
+		for _, a := range options.DefaultAttestors {
+			if name == a {
+				name = name + " (default)"
```
Comment (kriscoleman) [inline cmd/attestors.go:62]: also very helpful! 

I've heard a lot of questions about "why did this attestor run?"
This should help make it more intuitive

**Review by kriscoleman — APPROVED** [review]

**Comment by mikhailswift** [comment]
Definitely a very welcome addition and will become even more necessary as more attestors get added. I do wonder if it's worth giving the `command-run` attestor the same treatment as the `product` and `material` attestors.

Need to run `make docgen` to update the docs

**Comment by ChaosInTheCRD** [comment]
> Definitely a very welcome addition and will become even more necessary as more attestors get added. I do wonder if it's worth giving the `command-run` attestor the same treatment as the `product` and `material` attestors.
> 
> Need to run `make docgen` to update the docs

So @mikhailswift, you have probably noticed that I *deliberately* removed `command-run` from the list of "available attestors". This is for the simple fact that it's a slightly confusing user experience to have one attestor that simply isn't designed to be supplied with the `-a` flag. This is highlighted by https://github.com/in-toto/witness/issues/372 and @Strakeln's comments.

Maybe `command-run` simply *should* be listed here, and instead we should better handle the case where someone calls `-a command-run`. Possibly we should warn the user (like we already do in the case of e.g., `witness run -a product,product`, to inform them that the `command-run` attestor is invoked by supplying arguments to the witness command? I just think we need to put some thought into the user experience around this to make it more obvious.

**Comment by jkjell** [comment]
> Maybe command-run simply should be listed here

This is maybe a useless philosophical perspective for end users but, for me, `command-run` is implicit in the invocation of `witness run`. I think adding it as "always run" is a reasonable start to get feedback on.

**Comment by mikhailswift** [comment]
I see both sides and am good either way. I do think adding some logic to handle the case where `-a "command-run"` gets passed in and handled gracefully would be a very welcome QOL change, regardless of whether or not we add it to the attestor list here.

**Review by jkjell — APPROVED** [review]

**Pull request merged**